src/gui/MainWindow.cpp

```diff
@@ -1285,27 +1285,8 @@ void MainWindow::keyPressEvent( QKeyEvent * _ke )
 		case Qt::Key_Alt: m_keyMods.m_alt = true; break;
 		case Qt::Key_Space:
 		{
-			Editor* lastEditor = nullptr;
-			switch (Engine::getSong()->lastPlayMode())
-			{
-			case Song::PlayMode::Song:
-				lastEditor = getGUI()->songEditor();
-				break;
-			case Song::PlayMode::MidiClip:
-				lastEditor = getGUI()->pianoRoll();
-				break;
-			case Song::PlayMode::Pattern:
-				lastEditor = getGUI()->patternEditor();
-				break;
-			case Song::PlayMode::AutomationClip:
-				lastEditor = getGUI()->automationEditor();
-				break;
-			default:
-				lastEditor = getGUI()->songEditor();
-				break;
-			}
-			if (m_keyMods.m_shift) { lastEditor->togglePause(); }
-			else { lastEditor->togglePlayStop(); }
+			if (m_keyMods.m_shift) { Editor::s_lastPlayedEditor->togglePause(); }
+			else { Editor::s_lastPlayedEditor->togglePlayStop(); }
```

Comment:
Is there ever a case where `s_lastPlayedEditor` could be null? If you're not sure, better put a check or at least an `assert` here.